### PR TITLE
Content, layout config opt made available via command line

### LINKED ISSUE #1598: All location config options should be available via command line flags
See

https://discuss.gohugo.io/t/content-location-switch/1411

### PR DESCRIPTION
**Pull request by kujohn**
I want to get some feedback before working further.
fixes #1598

### CODE REVIEW

**Comment by kujohn** [comment]
Is this okay or should I instead add an additional argument to `helper.AbsPathify` for using `WorkingDir`? There's a todo comment [here](https://github.com/spf13/hugo/pull/1698/files#diff-d643be427d16a2670dc0a34546e82a50R133). A lot more files will be effected if so.

**Comment by spf13** [comment]
Please don't add an additional argument. The current approach you are taking is sound.

**Comment by spf13** [comment]
when you have this ready, please rebase from master and force push.

**Comment by kujohn** [comment]
:+1: ok

**Comment by kujohn** [comment]
Squashed and rebased, this should be good to go after code review.

**Comment by bep** [comment]
@kujohn could you rebase this against master ... again; sorry for that.

**Comment by kujohn** [comment]
Had some issue rebasing. Be back in an hour. :sob:

**Comment by kujohn** [comment]
Rebased correctly I think.

File: commands/hugo.go [hunk 1]
```diff
@@ -593,8 +605,15 @@ func copyStatic() error {
 // getDirList provides NewWatcher() with a list of directories to watch for changes.
 func getDirList() []string {
 	var a []string
+	var layoutPath string
+
+	if layoutDir != "" {
+		layoutPath = helpers.AbsPathifySimple(viper.GetString("LayoutDir"))
```
Comment (bep) [inline commands/hugo.go]: Why AbsPathifySimple? These new dir flags should behave exactly as the others. If there is some issue with AbsPathify, then that should be addressed in a separate issue.

Comment (kujohn) [inline commands/hugo.go]: If the working directory is set, this allows relative layout/content path to be set independently and working.
Perhaps this feature is not needed yet or that this func needs to be renamed?

**Comment by bep** [comment]
>Perhaps this feature is not needed yet or that this func needs to be renamed?

This is another issue not relevant to this issue. Remove that function and use the existing.

**Comment by kujohn** [comment]
ok squash and cleaned up.

**Comment by moorereason** [comment]
lgtm

**Comment by bep** [comment]
Looks good to me, too, thanks. Merged in 0ab4162

**Pull request closed without merging**